pkg/cli/admin/upgrade/recommend/recommend.go

```diff
@@ -142,6 +149,36 @@ func (o *options) Run(ctx context.Context) error {
 		fmt.Fprintf(o.Out, "info: An update is in progress.  You may wish to let this update complete before requesting a new update.\n  %s\n", strings.ReplaceAll(c.Message, "\n", "\n  "))
 	}
 
+	if o.precheckEnabled {
+		conditions, err := o.precheck(ctx)
+		if err != nil {
+			fmt.Fprintf(o.Out, "Failed to check for at least some preconditions: %v\n", err)
+		}
+		var happyConditions []string
+		var unhappyConditions []string
+		for _, condition := range conditions {
+			if condition.Status == metav1.ConditionTrue {
```

Comment:
Here we make an assumption that all conditions returned by `o.precheck` are positive (`Status=True` is a happy thing) which is not a general contract for conditions (like Failing=True or Degraded=True). Can we add a godoc for `o.precheck` that will describe this expectation, for whoever adds new sources of conditions to be returned there?

Reply:
Godocs added to `precheck` and `alerts` as part of the 426959c86 -> 4d19abdf7 reroll, where I also pivoted the `NoTestData` condition from `True` to `Unknown`.